lib/Agrammon/Model.pm6

```diff
@@ -516,4 +535,45 @@ class Agrammon::Model {
         };
         %$!distribution-map-cache
     }
+
+    #| Get a CSV formatted template of all model inputs
+    #| with one demo instance for all multi-instance modules.
+    method get-input-template($sort) {
+        # Obtain model structure in specified order.
+        my @model-structure := self.extract-structure-sorted($sort);
+        # Go through the model structure.
+        my Str @inputs;
+        for @model-structure {
+            when Str {
+                # Single instance module
+                my $module = self.get-module($_);
+                add-module-inputs(@inputs, $module);
+            }
+            when Pair {
+                # Multi-instance module
+                my @modules = @(.value);
+                my $instance-name = 'Demo';
+                for @modules -> $module-name is copy {
+                    my $module = self.get-module($module-name);
+                    add-module-inputs(@inputs, $module, :$instance-name);
+                }
+            }
+        }
```

Comment:
This lot can be structured as:

```
for @modules -> $module {
    if $module.instance-root -> $root {
        # Multi-instance module
        ...
    }
    else {
        # Single-instance module
        ...
    }
}
```

Note that since you don't have strings but module objects, there's no need to redo the module lookup either.